tests/unit/dataactbroker/test_file_handler.py

```diff
@@ -120,3 +124,87 @@ def test_list_submissions_failure(database, job_constants, monkeypatch):
     assert json.loads(json_response.get_data().decode("utf-8"))['total'] == 1
     assert json.loads(json_response.get_data().decode("utf-8"))['submissions'][0]['status'] == "file_errors"
     delete_models(database, [user, sub, job])
+
+
+def test_requires_submission_perm_no_submission(database, monkeypatch):
+    """If no submission exists, we should see an exception"""
+    monkeypatch.setattr(fileHandler, 'user_agency_matches',
+                        Mock(return_value=True))
+
+    sub = SubmissionFactory()
+    database.session.add(sub)
+    database.session.commit()
+
+    fn = fileHandler.requires_submission_perms(Mock())
+    # Does not raise exception
+    fn(sub.submission_id)
+    with pytest.raises(ResponseException):
+        fn(sub.submission_id + 1)   # different submission id
+
+
+def test_requires_submission_perm_check(database, monkeypatch):
+    """If the user doesn't have a matching agency, we should see an
+    exception"""
+    sub = SubmissionFactory()
+    database.session.add(sub)
+    database.session.commit()
+
+    fn = fileHandler.requires_submission_perms(Mock())
+    # Does not raise exception
+    monkeypatch.setattr(fileHandler, 'user_agency_matches',
+                        Mock(return_value=True))
+    fn(sub.submission_id)
+
+    monkeypatch.setattr(fileHandler, 'user_agency_matches',
+                        Mock(return_value=False))
+    with pytest.raises(ResponseException):
+        fn(sub.submission_id)
+
+
+def test_narratives(database, job_constants, monkeypatch):
+    """Integration test that verifies we can add, retrieve, and update
```

Comment:
Eh, I don't have clear definitions, so I'm probably using the term incorrectly. I meant "this isn't testing a single unit of work, it's testing a few at once". I can swap out to be more explicit if that'd be better?